source/module_rdmft/rdmft_tools.h

```diff
@@ -0,0 +1,429 @@
+//==========================================================
+// Author: Jingang Han
+// DATE : 2024-03-11
+//==========================================================
+#ifndef RDMFT_TOOLS_H
+#define RDMFT_TOOLS_H
+
+#include "module_psi/psi.h"
+#include "module_base/matrix.h"
+#include "module_cell/module_neighbor/sltk_grid_driver.h"
+#include "module_cell/unitcell.h"
+#include "module_hamilt_lcao/module_gint/gint_gamma.h"
+#include "module_hamilt_lcao/module_gint/gint_k.h"
+#include "module_elecstate/potentials/potential_new.h"
+#include "module_base/blas_connector.h"
+#include "module_base/scalapack_connector.h"
+#include "module_basis/module_ao/parallel_2d.h"
+#include "module_basis/module_ao/parallel_orbitals.h"
+#include "module_base/parallel_reduce.h"
+#include "module_hamilt_pw/hamilt_pwdft/global.h"
+#include "module_elecstate/module_dm/cal_dm_psi.h"
+#include "module_elecstate/module_dm/density_matrix.h"
+
+#include "module_hamilt_lcao/module_hcontainer/hcontainer.h"
+#include "module_hamilt_lcao/hamilt_lcaodft/hs_matrix_k.hpp"
+#include "module_hamilt_lcao/hamilt_lcaodft/operator_lcao/operator_lcao.h"
+
+
+#ifdef __EXX
+#include "module_ri/RI_2D_Comm.h"
+#include "module_ri/Exx_LRI.h"
+// there are some operator reload to print data in different formats
+#include "module_ri/test_code/test_function.h"
+#endif
+
+#include <iostream>
+#include <type_traits>
+#include <complex>
+#include <vector>
+#include <iomanip>
+
+
+
+namespace rdmft
```

Comment:
this is quite a good practice compared with other codes because you use namespace instead of any absurd class to place tool functions.